File: java/speech-expenses/BasicAuth.java
import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.nio.file.Files;
import java.util.Base64;

public class BasicAuth {
    public static void main(String args[]) {
        // basic configuration
        String clientId = "[[client_id]]";
        String clientSecret = "[[client_secret]]";

        // create a base64 encoded api key
        String apiKey = Base64.getEncoder().encodeToString((clientId.concat(":")).concat(clientSecret).getBytes());

        // use the api key as a basic token
        String authorization = "Basic ".concat(apiKey);

        // populate endpoint parameters
        String fullPathToFile = "[[filepath]]";
        String userAgent = "Intelligent API Sample Java Code";

        // invoke the API endpoint
        try {
            File file = new File(fullPathToFile);
            String boundary = "----FormBoundary" + System.currentTimeMillis();

            // Build multipart body
            ByteArrayOutputStream bodyStream = new ByteArrayOutputStream();
            PrintWriter writer = new PrintWriter(new OutputStreamWriter(bodyStream, "UTF-8"), true);

            writer.append("--").append(boundary).append("\r\n");
            writer.append("Content-Disposition: form-data; name=\"file\"; filename=\"")
                    .append(file.getName())
                    .append("\"").append("\r\n");
            writer.append("Content-Type: application/octet-stream").append("\r\n");
            writer.append("\r\n");
            writer.flush();

            bodyStream.write(Files.readAllBytes(file.toPath()));

            writer.append("\r\n");
            writer.append("--").append(boundary).append("--").append("\r\n");
            writer.flush();

            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create("https://api.intelligent-api.com/v1/speech/expenses?classifications=Food%20%26%20Dining,Transport,Shopping,Utilities,Entertainment,Health,Education,Other"))
                    .header("Authorization", authorization)
                    .header("User-Agent", userAgent)
                    .header("Content-Type", "multipart/form-data; boundary=" + boundary)
                    .POST(HttpRequest.BodyPublishers.ofByteArray(bodyStream.toByteArray()))
                    .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            System.out.println("Response Code: " + response.statusCode());
            System.out.println("Response Body: " + response.body());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: python/speech-expenses/basic-auth.py
import base64
import http.client
import mimetypes
import os
import uuid

# basic configuration
client_id = "[[client_id]]"
client_secret = "[[client_secret]]"

# create a base64 encoded api key
api_key_bytes = base64.b64encode((client_id + ":" + client_secret).encode("utf-8"))
api_key = api_key_bytes.decode("utf-8")

# use the api key as a basic token
authorization = "Basic " + api_key

# populate endpoint parameters
file_path = "[[filepath]]"
user_agent = "Intelligent API Sample Python Code"

# build multipart form body
boundary = uuid.uuid4().hex
body = []

# form part headers
body.append(f"--{boundary}".encode())
body.append(
    f'Content-Disposition: form-data; name="file"; filename="{os.path.basename(file_path)}"'.encode()
)
mime_type = mimetypes.guess_type(file_path)[0] or "application/octet-stream"
body.append(f"Content-Type: {mime_type}".encode())
body.append(b"")

# file content
with open(file_path, "rb") as file:
    body.append(file.read())

# closing boundary
body.append(f"--{boundary}--".encode())

multipart_body = b"\r\n".join(body)

# invoke the API endpoint
host = "api.intelligent-api.com"
path = "/v1/speech/expenses?classifications=Food%20%26%20Dining,Transport,Shopping,Utilities,Entertainment,Health,Education,Other"

connection = http.client.HTTPSConnection(host)

headers = {
    "Authorization": authorization,
    "Content-Type": f"multipart/form-data; boundary={boundary}",
    "User-Agent": user_agent,
}

connection.request("POST", path, body=multipart_body, headers=headers)
response = connection.getresponse()
response_data = response.read().decode("utf-8")

print("Status Code:", response.status)
print("Response:", response_data)

connection.close()
